packages/mui-utils/src/useForkRef/useForkRef.ts

```diff
@@ -20,6 +20,7 @@ export default function useForkRef<Instance>(
         setRef(ref, instance);
       });
     };
+    // TODO: uncomment once we enable eslint-plugin-react-compiler // eslint-disable-next-line react-compiler/react-compiler -- dependency must be array
```

Comment:
Maybe expand the explanation a bit
```suggestion
    // TODO: uncomment once we enable eslint-plugin-react-compiler // eslint-disable-next-line react-compiler/react-compiler -- intentionally ignoring that the dependency array must be an array literal
```